modules/exploits/linux/local/bash_profile_persistence.rb

```diff
@@ -0,0 +1,112 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Local
+  Rank = NormalRanking
+  include Msf::Post::Common
+  include Msf::Post::File
+  include Msf::Post::Unix
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name'           => 'Bash Profile Persistence',
+        'Description'    => "
+          This module writes an execution trigger to the target's Bash profile.
+          The execution trigger executes a call back payload whenever the target
+          user opens a Bash terminal. A handler is not run automatically, so you
+          must configure an appropriate exploit/multi/handler to receive the callback.
+        ",
+        'License'        => MSF_LICENSE,
+        'Author'         =>
+          [
+            'Michael Long <bluesentinel[at]protonmail.com>'
+          ],
+        'DisclosureDate' => 'Jun 8 1989', # First public release of Bourne Again Shell
+        'Platform'       => ['unix', 'linux'],
+        'Arch'           => ARCH_CMD,
+        'SessionTypes'   => ['meterpreter'],
```

Comment:
Any reason it's just meterpreter?

Reply:
Presently the module throws an error when executed in generic shells:
`[-] Exploit failed: NoMethodError undefined method `sys' for #<Session:shell 127.0.0.1:34166 (127.0.0.1) >`
I think its being caused by these lines:
```
home_directory = session.sys.config.getenv('$HOME')
...
hostname = session.sys.config.sysinfo["Computer"]
```

Reply:
Yep that would be it. If you can find an alternative way to get the home directory (say by executing a shell command or something else) that should make it work on shell sessions as well.

Reply:
I added support for standard shell sessions.
If the Meterpreter API method fails, the module runs shell commands to pull the user's home directory and host name.